### ISSUE TITLE
Unable te install MrBayes on Apple M1 from source

### ISSUE THREAD

**Issue by vresbok** [NBISweden/MrBayes#248]
I am trying to install MrBayes from source and with the `./configure --disable-{sse,avx,fma}`
, after having the same problem as issue #241 "Unable te install MrBayes on Apple M1 hardware (arm CPU issue?)" when I tried installing with homebrew.
However, at the `make` command I receive the below error. 

Thanks in advance for any help with this issue. 

### What is the current observed behaviour?
MacBook-Pro MrBayes % ./configure --disable-{sse,avx,fma}
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... autotools/install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking build system type... arm-apple-darwin20.4.0
checking host system type... arm-apple-darwin20.4.0
checking for pkg-config... /opt/homebrew/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for pdflatex... /Library/TeX/texbin/pdflatex
checking for bibtex... /Library/TeX/texbin/bibtex
checking for latexmk... /Library/TeX/texbin/latexmk
checking for style of include used by make... GNU
checking whether to compile using MPI... no
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for sqrt in -lm... yes
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
configure: Trying to locate Beagle (hmsbeagle-1) in '/usr/local'
checking libhmsbeagle/beagle.h usability... no
checking libhmsbeagle/beagle.h presence... no
checking for libhmsbeagle/beagle.h... no
checking for beagleCreateInstance in -lhmsbeagle... no
configure: Trying to locate Beagle (hmsbeagle-1) in '/usr'
checking libhmsbeagle/beagle.h usability... no
checking libhmsbeagle/beagle.h presence... no
checking for libhmsbeagle/beagle.h... no
checking for beagleCreateInstance in -lhmsbeagle... (cached) no
configure: WARNING: Failed to locate Beagle, falling back on pkg-config
checking for BEAGLE... no
checking for a readline compatible library... -lreadline
checking readline.h usability... no
checking readline.h presence... no
checking for readline.h... no
checking readline/readline.h usability... yes
checking readline/readline.h presence... yes
checking for readline/readline.h... yes
checking whether readline supports history... yes
checking history.h usability... no
checking history.h presence... no
checking for history.h... no
checking readline/history.h usability... yes
checking readline/history.h presence... yes
checking for readline/history.h... yes
checking float.h usability... yes
checking float.h presence... yes
checking for float.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking for memory.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for long long... yes
checking for size_t... yes
checking for stdbool.h that conforms to C99... yes
checking for _Bool... yes
checking for C compiler vendor... clang
checking for a sed that does not truncate output... /usr/bin/sed
checking for C compiler version... 12.0.5
checking CFLAGS for maximum warnings... -Wall
checking for x86 cpuid  output... unknown
checking for x86-AVX xgetbv  output... unknown
checking for floor... yes
checking for memset... yes
checking for pow... yes
checking for sqrt... yes
checking for strchr... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating doc/Makefile
config.status: creating doc/SMME_paper/Makefile
config.status: creating doc/manual/Makefile
config.status: creating examples/Makefile
config.status: creating src/Makefile
config.status: creating testing/Makefile
config.status: creating testing/runtests.sh
config.status: creating src/config.h
config.status: executing depfiles commands

MacBook-Pro MrBayes % make
Making all in src
/Library/Developer/CommandLineTools/usr/bin/make  all-am
gcc -DHAVE_CONFIG_H -I.       -O3 -DNDEBUG -std=c99 -pedantic  -Wall -MT mb-bayes.o -MD -MP -MF .deps/mb-bayes.Tpo -c -o mb-bayes.o `test -f 'bayes.c' || echo './'`bayes.c
mv -f .deps/mb-bayes.Tpo .deps/mb-bayes.Po
gcc -DHAVE_CONFIG_H -I.       -O3 -DNDEBUG -std=c99 -pedantic  -Wall -MT mb-best.o -MD -MP -MF .deps/mb-best.Tpo -c -o mb-best.o `test -f 'best.c' || echo './'`best.c
mv -f .deps/mb-best.Tpo .deps/mb-best.Po
gcc -DHAVE_CONFIG_H -I.       -O3 -DNDEBUG -std=c99 -pedantic  -Wall -MT mb-command.o -MD -MP -MF .deps/mb-command.Tpo -c -o mb-command.o `test -f 'command.c' || echo './'`command.c
mv -f .deps/mb-command.Tpo .deps/mb-command.Po
gcc -DHAVE_CONFIG_H -I.       -O3 -DNDEBUG -std=c99 -pedantic  -Wall -MT mb-likelihood.o -MD -MP -MF .deps/mb-likelihood.Tpo -c -o mb-likelihood.o `test -f 'likelihood.c' || echo './'`likelihood.c
mv -f .deps/mb-likelihood.Tpo .deps/mb-likelihood.Po
gcc -DHAVE_CONFIG_H -I.       -O3 -DNDEBUG -std=c99 -pedantic  -Wall -MT mb-mbbeagle.o -MD -MP -MF .deps/mb-mbbeagle.Tpo -c -o mb-mbbeagle.o `test -f 'mbbeagle.c' || echo './'`mbbeagle.c
mv -f .deps/mb-mbbeagle.Tpo .deps/mb-mbbeagle.Po
gcc -DHAVE_CONFIG_H -I.       -O3 -DNDEBUG -std=c99 -pedantic  -Wall -MT mb-mcmc.o -MD -MP -MF .deps/mb-mcmc.Tpo -c -o mb-mcmc.o `test -f 'mcmc.c' || echo './'`mcmc.c
mcmc.c:5959:16: error: no member named 'numFloatsPerVec' in 'struct modelinfo'
            m->numFloatsPerVec = 0;
            ~  ^
1 error generated.
make[2]: *** [mb-mcmc.o] Error 1
make[1]: *** [all] Error 2
make: *** [all-recursive] Error 1

MacBook-Pro MrBayes %

### What is the expected/wanted behaviour?
Successful installation.

### How may we reproduce this bug?
Try to install from source on Mac M1.

### Would you be able to compile and run MrBayes to test fixes to this bug?
  - [ ] Yes, I know how to use `git` and how to compile MrBayes.
  - [ ] No, sorry.

### What is the environment that you run MrBayes in?
  - Operating system (including variant and release): Mac OSX 11.3 (M1)
  - Version of MrBayes: Trying to install MrBayes 3.2.7a

**Comment by andkaha** [comment]
Thanks for finding this bug!  I believe that I have fixed it in our development branch.

Try checking out the `develop` branch and running `configure` and `make` again.

The issue was due to a piece of code that evidently does not see much action. It should have been disabled when the Beagle library isn't available (it wasn't disabled out, but now it is).

We currently do not have access to any M1 Apple hardware, so we can't unfortunately not test this ourselves at the moment.
